### DISCUSSION TITLE
What is Missing or Grinding Your Gears?

### DISCUSSION THREAD

**Discussion by JAAdrian** [JAAdrian/MatlabProgressBar#10] [Ideas]
Hi all,

I would really like to hear some feedback of you, the users of the progress bar. Is there something which is really bugging you and/or not functioning the way you want?

Often there are technical limitations why things are the way they are but sometimes it's just really a lack of a fresh perspective.

So, fire away!

**Comment by tunakasif** [comment]
Hey @JAAdrian, thanks for the great project 👍🏻 you made writing MATLAB code bearable; it would be great if you could also provide an example for nested loops where the inner loop is parallel, i.e., `parfor`. Just replacing the [parallel example](https://github.com/JAAdrian/MatlabProgressBar/blob/5e9e29445aeb2a640af5719e6bdd4f814584a34c/demos/parallelSetup.m#L41-L55), with the inner loop of the [nested example](https://github.com/JAAdrian/MatlabProgressBar/blob/5e9e29445aeb2a640af5719e6bdd4f814584a34c/demos/nestedProgressBars.m#L24-L35) creates empty inner loop line, seems like an artifact of `setup()`. If you can provide a correct way of making nested loops with inner `parfor`, that would be great 🎉. I have attached my attempt below.

```matlab
numOuterIterations = 3;
numInnerIterations = 15;
b1 = ProgressBar(numOuterIterations, 'Title', 'Loop 1');
b1.setup([], [], []);
for iOuterIteration = 1:numOuterIterations
    b2 = ProgressBar(numInnerIterations, ...
        'IsParallel', true, ...
        'WorkerDirectory', pwd(), ...
        'Title', 'Loop 2' ...
    );
    b2.setup([], [], []);
    parfor jInnerIteration = 1:numInnerIterations
        pause(0.1);
        updateParallel([], pwd);
    end
    b2.release();
    b1([], [], []);
end
b1.release();
ProgressBar.deleteAllTimers();
```

As you can see, on the right-hand side of the outer loop, there are empty lines of the inner loop.

![Screenshot from 2023-08-27 00-22-11](https://github.com/JAAdrian/MatlabProgressBar/assets/34691280/f8e6e4c8-a859-4b23-94a5-6091b83a9980)
